src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -582,12 +582,23 @@ private static ReturnCode displaySummary(int fileCount, int invalidCount, int sk
     }
 
     /**
-     * Tells the validator to monitor progress and show ongoing validation progress
+     * Tells the validator whether or not to monitor progress and show ongoing validation progress
      *
-     * @param id an identifier for the progress monitor
+     * @param id if non-null, an identifier for the progress monitor, otherwise disables progress monitoring
      */
     public void setProgressMonitor(String id) {
-        this.monitorID = id;
+        if (id == null) {
+            this.progressMonitor = null;
+            return;
+        }
+
+        final String filename = id + "-progress.tab";
+        try {
+            this.progressMonitor = new AIFProgressMonitor("");
```

Comment:
Don't you mean to pass filename here?  This fails with an IOException saying it can't write to filename.

Reply:
Whoops-- I forgot to remove that after testing proper error handling-- good catch!